src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpRequestHeaders.cs

```diff
@@ -159,6 +159,7 @@ private T GetSpecializedCollection<T>(int slot, Func<HttpRequestHeaders, T> crea
             set { SetOrRemoveParsedValue(KnownHeaders.MaxForwards.Descriptor, value); }
         }
 
+        public string? Protocol { get; set; }
```

Comment:
We should do the `CheckContainsNewLine` validation as part of the setter here as we do for other headers.

Reply:
Right, I moved it here from `Http2Connection`